src/main/java/util/JsonHelper.java

```diff
@@ -0,0 +1,89 @@
+ */
+public class JsonHelper {
+    private static final Logger logger = LogManager.getLogger(JsonHelper.class.getName());
+
+    private static final String EMPTY_STRING = "";
+
+    private Gson gson;
+
+    /**
+     * Sets up the JsonHelper by initialising the Gson component
+     */
+    public JsonHelper() {
+        setupGson();
+    }
+
+    /**
+     * Configures the gson object that will serialise and deserialise json data
+     */
+    private void setupGson() {
+        gson = new GsonBuilder()
+                .setPrettyPrinting()
+                .registerTypeAdapter(LocalDateTime.class,
+                        (JsonSerializer<LocalDateTime>) (src, typeOfSrc, context) -> {
+                            Instant instant = src.atZone(ZoneId.systemDefault()).toInstant();
+                            long epochMilli = instant.toEpochMilli();
+                            return new JsonPrimitive(epochMilli);
+                        })
+                .registerTypeAdapter(LocalDateTime.class,
+                        (JsonDeserializer<LocalDateTime>) (json, typeOfT, context) -> {
+                            Instant instant = Instant.ofEpochMilli(json.getAsJsonPrimitive().getAsLong());
+                            return LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+                        }
+                ).create();
+    }
+
+    /**
+     * Creates a specific Config subclass instance from a JSON-formatted String
+     * @param jsonString The String in JSON format that contains the Config values
+     * @param configClass The class of Config to create from the JSON
+     * @return The Config instance of the specified class
+     */
+    public Config createConfigFromJson(String jsonString, Class<? extends Config> configClass) {
+        if (jsonString.equals(EMPTY_STRING)) {
+            try {
+                logger.warn("Empty string json supplied, creating new Config instance");
+                return configClass.newInstance();
+
+            } catch (InstantiationException | IllegalAccessException e) {
+                HTLog.error(logger, e);
+                logger.error("Unable to create new Config instance! Using gson to create from empty string instead.");
+                return gson.fromJson(jsonString, configClass);
+            }
+        }
+        Config createdConfig = gson.fromJson(jsonString, configClass);
+        logger.info("Config successfully created");
+        return createdConfig;
+    }
+
+    /**
+     * Creates a JSON-formatted string from the values in a Config instance
+     * @param config The Config object to be converted into the JSON string
+     * @return The JSON-formatted string that contains the values of the given Config
+     */
+    public String createJsonFromConfig(Config config) {
+        if (config == null) {
+            logger.warn("Null config supplied, returning empty string");
+            return EMPTY_STRING;
```

Comment:
That should be handled at that point, then. If we know that deserialisation failed earlier, we shouldn't be calling this method.